design/009-pushsecret-generator.md

```diff
@@ -0,0 +1,62 @@
+```yaml
+---
+title: PushSecret generator integration
+version: v1alpha1
+authors: Moritz Johner
+creation-date: 2023-08-25
+status: draft
+---
+```
+
+# PushSecret Generator integration
+
+## Table of Contents
+
+<!-- toc -->
+// autogen please
+<!-- /toc -->
+
+## Summary
+
+This design document describes how `PushSecret` can leverage generators to generate short-lived credentials without the need of an intermediary `Secret` resource.
+
+## Motivation
+
+Currently, the process of using secure passwords and short-lived credentials within the External Secrets Operator involves multiple steps.
+Users need to create an `ExternalSecret` resource to generate a value which is stored in a `Secret` resource. This Secret resource is then pushed to a provider using a `PushSecret` resource. However, this intermediary step adds unnecessary complexity and inconvenience to the workflow.
+
+## Proposal
+
+To simplify the workflow and enhance user experience, the proposal is to integrate generators directly into the `PushSecret` resource. This will allow users to specify a generator using a `generatorRef` within the PushSecret manifest. When the PushSecret reconciliation process occurs, a value will be generated using the specified generator. This generated value will be securely pushed to the provider and stored there.
+
+```yaml
+apiVersion: external-secrets.io/v1alpha1
+kind: PushSecret
+spec:
+  selector:
+    generatorRef:
+      apiVersion: generators.external-secrets.io/v1alpha1
+      kind: Password
+      name: "my-password"
```

Comment:
IIUC, users need to configure `data.match.secretKey` manually to select which secret to push upstream, correct? 🤔 

```
  data:
    - match:
        secretKey: password
        remoteRef:
          remoteKey: some-remote-key
```

Reply:
yes, they have to individually specify the secretKey to select what to push upstream. Once we have PushSecret templates, the user can better define how to structure of the secret.